app/presenters/openid_connect_user_info_presenter.rb

```diff
@@ -20,11 +20,15 @@ def user_info
     }
 
     info[:all_emails] = all_emails_from_sp_identity(identity) if scoper.all_emails_requested?
-    info.merge!(ial2_attributes) if scoper.ial2_scopes_requested? && ial2_data.present?
+    info.merge!(ial2_attributes) if identity_proofing_requested_for_verified_user?
     info.merge!(x509_attributes) if scoper.x509_scopes_requested?
     info[:verified_at] = verified_at if scoper.verified_at_requested?
     if identity.vtr.nil?
-      info[:ial] = Saml::Idp::Constants::AUTHN_CONTEXT_IAL_TO_CLASSREF[identity.ial]
+      info[:ial] = if identity_proofing_requested_for_verified_user?
+                     Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF
+                   else
+                    Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF
+                   end
```

Comment:
indentation is weird here?
```suggestion
      info[:ial] = if identity_proofing_requested_for_verified_user?
                     Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF
                   else
                     Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF
                   end
```